src/datastore/pg-write-store.ts

```diff
@@ -2450,6 +2449,16 @@ export class PgWriteStore extends PgStore {
     if (this.isEventReplay && skipDuringEventReplay) {
       return;
     }
+    if (process.env.IBD_MODE_UNTIL_BLOCK) {
+      const db = new PgWriteStore(sql);
+      const chainTip = await db.getChainTip(sql);
+      if (
+        process.env.IBD_MODE_UNTIL_BLOCK &&
```

Comment:
This check is already done in the outer `if`, you can remove it here

Reply:
done